### PR TITLE
Follow naming convention in partitioned heat conduction - FEniCS

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Fixed incorrect order for OpenFOAM

Updated structure to the one provided [here](https://precice.org/community-contribute-to-precice.html#structure-of-a-tutorial) for FEniCS.

Other cases are still missing.

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@MakisH I think the order in the OpenFOAM cases was incorrect. Fixed it. For the other cases we also need an update, but maybe this is also a good first issue? Should be relatively simple to do and is still a valuable contribution.

File: partitioned-heat-conduction/solver-fenics/heat.py [hunk 1]
```diff
@@ -101,15 +99,14 @@ def determine_gradient(V_g, u, flux):
...
     precice.initialize(coupling_boundary, read_function_space=V, write_object=f_N_function)
-    precice_dt = precice.get_max_time_step_size()
 elif problem is ProblemType.NEUMANN:
-    precice = Adapter(adapter_config_filename="precice-adapter-config-N.json")
     precice.initialize(coupling_boundary, read_function_space=W, write_object=u_D_function)
-    precice_dt = precice.get_max_time_step_size()
 
+precice_dt = precice.get_max_time_step_size()
 dt = Constant(0)
 dt.assign(np.min([fenics_dt, precice_dt]))
```
Comment (BenjaminRodenberg) [inline partitioned-heat-conduction/solver-fenics/heat.py:111]: I like the fact that the new structure simplifies the mapping of solver to config file :+1:

**Review by MakisH — APPROVED** [review]
Thank you for fixing this! I will update the structure specification as well.

The `openfoam-solver` was there before we come up with some kind of naming convention for solver directories. Thank you for fixing that as well!

Changes look good. This will not create conflicts with any other PR, right?

edit: Not an issue (but PR), therefore also not a good-first-issue.

**Comment by BenjaminRodenberg** [comment]
> Changes look good. This will not create conflicts with any other PR, right?

No. Should be fine.

> edit: Not an issue (but PR), therefore also not a good-first-issue.

Do you think we can find somebody to take care about the rest? It's a nice-to-have and currently don't want to prioritize this. So let's either take care about all cases soon or directly merge and take care about the rest later (feel free to merge).

**Comment by MakisH** [comment]
> > edit: Not an issue (but PR), therefore also not a good-first-issue.
> 
> Do you think we can find somebody to take care about the rest? It's a nice-to-have and currently don't want to prioritize this. So let's either take care about all cases soon or directly merge and take care about the rest later (feel free to merge).

Let's open an issue for that, with a checklist of which cases need to be restructured.

I would already merge this one, to let other PRs come as well.

**Comment by MakisH** [comment]
I consider this ready to merge. I prepared an overview regarding the rest of the cases here: https://github.com/precice/tutorials/issues/461

**Review by MakisH — APPROVED** [review]

**Comment by davidscn** [comment]
With the changes applied here and in #477 the `solver-openfoam` doesn't compile anymore. Did anyone test this before merging or is it just a local problem of my OpenFOAM installation? (tried with 2306 and 2012.

**Comment by MakisH** [comment]
I don't see how it could be at all an issue of the renaming, but it indeed does not compile for me anymore:

```
~/repos/precice/tutorials/partitioned-heat-conduction/solver-openfoam [develop]$ wmake
Making dependencies: heatTransfer.C
g++ -std=c++14 -m64 -pthread -DOPENFOAM=2312 -DWM_DP -DWM_LABEL_SIZE=32 -Wall -Wextra -Wold-style-cast -Wnon-virtual-dtor -Wno-unused-parameter -Wno-invalid-offsetof -Wno-attributes -Wno-unknown-pragmas -O3  -DNoRepository -ftemplate-depth-100  -I/usr/lib/openfoam/openfoam2312/src/finiteVolume/lnInclude -I/usr/lib/openfoam/openfoam2312/src/meshTools/lnInclude -iquote. -IlnInclude -I/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude -I/usr/lib/openfoam/openfoam2312/src/OSspecific/POSIX/lnInclude   -fPIC -c heatTransfer.C -o Make/linux64GccDPInt32Opt/heatTransfer.o
In file included from heatTransfer.C:89:
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H: In function ‘int main(int, char**)’:
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H:37:5: error: ‘args’ was not declared in this scope; did you mean ‘argc’?
   37 |     args.getOrDefault<word>("region", Foam::polyMesh::defaultRegion)
      |     ^~~~
      |     argc
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H:37:27: error: expected primary-expression before ‘>’ token
   37 |     args.getOrDefault<word>("region", Foam::polyMesh::defaultRegion)
      |                           ^
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H:37:29: warning: left operand of comma operator has no effect [-Wunused-value]
   37 |     args.getOrDefault<word>("region", Foam::polyMesh::defaultRegion)
      |                             ^~~~~~~~
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H:51:50: error: ‘runTime’ was not declared in this scope
   51 |         new Foam::simplifiedMeshes::columnFvMesh(runTime, regionName)
      |                                                  ^~~~~~~
/usr/lib/openfoam/openfoam2312/src/OpenFOAM/lnInclude/createMesh.H:81:37: error: ‘runTime’ was not declared in this scope
   81 |     Foam::Info << " for time = " << runTime.timeName() << Foam::nl;
      |                                     ^~~~~~~
make: *** [/usr/lib/openfoam/openfoam2312/wmake/rules/General/transform:38: Make/linux64GccDPInt32Opt/heatTransfer.o] Error 1
```

I don't remember when was the last time I tested this case, probably not when merging this, given that I needed to restore the `Make/` directory in #477.

**Comment by MakisH** [comment]
Irrelevant to this PR, but to https://github.com/precice/tutorials/pull/468. Fixed in https://github.com/precice/tutorials/pull/499.

**Pull request merged**